### ISSUE TITLE
youtube.com: Unable to extract Initial JS player n function name

### ISSUE THREAD

**Issue by nbraud** [ytdl-org/youtube-dl#30958]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```shell
$ youtube-dl --verbose 'https://youtu.be/-ZReLaWESAE'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://youtu.be/-ZReLaWESAE']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Lazy loading extractors enabled
[debug] Python version 3.9.12 (CPython) - Linux-5.15.37-x86_64-with-glibc2.34
[debug] exe versions: ffmpeg 4.4.1, ffprobe 4.4.1, rtmpdump 2.4
[debug] Proxy map: {}
[youtube] -ZReLaWESAE: Downloading webpage
[youtube] -ZReLaWESAE: Downloading player 9c7ce883
ERROR: -ZReLaWESAE: Traceback (most recent call last):
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1442, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1415, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1396, in _extract_n_function_name
    nfunc_and_idx = self._search_regex(
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
 (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1442, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1415, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1396, in _extract_n_function_name
    nfunc_and_idx = self._search_regex(
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1442, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1415, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1396, in _extract_n_function_name
    nfunc_and_idx = self._search_regex(
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1874, in _real_extract
    self._unthrottle_format_urls(video_id, player_url, formats)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1459, in _unthrottle_format_urls
    n_response = self._n_descramble(n_param, player_url, video_id)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1449, in _n_descramble
    raise ExtractorError(traceback.format_exc(), cause=e, video_id=video_id)
youtube_dl.utils.ExtractorError: -ZReLaWESAE: Traceback (most recent call last):
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1442, in _n_descramble
    self._player_cache[player_id] = self._extract_n_function(video_id, player_url)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1415, in _extract_n_function
    funcname = self._extract_n_function_name(jscode)
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/youtube.py", line 1396, in _extract_n_function_name
    nfunc_and_idx = self._search_regex(
  File "/nix/store/57cd98f35vimmkp7fn8jz07da83frl79-python3.9-youtube-dl-2021.12.17/lib/python3.9/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract Initial JS player n function name; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
 (caused by RegexNotFoundError('Unable to extract \x1b[0;34mInitial JS player n function name\x1b[0m; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

Attempting to download YouTube videos, all result in the same error (see above), regardless of whether `youtu.be` or `youtube.com/watch` links are used.  See error log above.

I was able to reproduce the issue on multiple, unrelated videos, which are all playable in Firefox (and in a clean “private browsing” tab)

**Comment by dirkf** [comment]
There's no n-descrambling in the released 2021.12.17 version.

Your yt-dl must be an older patched version. You need to [update to the latest git master](https://github.com/ytdl-org/youtube-dl/issues/30871#issuecomment-1102666158).

Duplicate of #30856.

**Comment by dirkf** [comment]
And to the absolute latest version: see #30976.
